snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -286,16 +286,23 @@ def _setup_snapcraft(self) -> None:
             file_pusher=self._push_file,
             inject_from_host=inject_from_host,
         )
+
+        # If the build base is core, we not need to inject snapd.
+        # Check for None as this build can be driven from a non snappy enabled
+        # system, so we may find ourselves in a situation where the base is not
+        # set like on OSX or Windows.
+        build_base = self.project.info.get_build_base()
+        if build_base is not None and build_base != "core":
+            snap_injector.add(snap_name="snapd")
+
+        # Prevent injecting core18 twice.
+        if build_base is not None and build_base != "core18":
+            snap_injector.add(snap_name=build_base)
```

Comment:
I'd expect the injector to dedupe fwiw

Reply:
But should I also change to append as order matters?
Is an append that may not append good language?